src/licensedcode/plugin_license.py

```diff
@@ -116,3 +119,57 @@ def get_scanner(
             license_text_diagnostics=license_text_diagnostics,
             license_url_template=license_url_template
         )
+
+    def process_codebase(self, codebase, **kwargs):
+        
+        if codebase.has_single_resource:
+            return
+
+        for resource in codebase.walk(topdown=False):
+            match_reference_license(resource,codebase)
+
+
+def match_reference_license(resource, codebase):
+    """
+    Find instances for any licenses in referenced filenames
+    """
+    if not resource.is_file:
+        return
+
+    licenses = resource.licenses
+    license_expressions = resource.license_expressions
+    if not licenses:
+        return 
+
+    reference_licenses=[]
+    reference_license_expressions=[]
+    unique_files = []
+
+    for license in licenses:
+        ref_files=license['matched_rule']['referenced_filenames']
+        if len(ref_files) != 0:
+            for i in range(len(ref_files)):
+                if ref_files[i].startswith('usr/share/common-licenses') or ref_files[i] in unique_files:
+                    continue
+                unique_files.append(ref_files[i])
+                new_resource = find_reference_licenses(ref_files[i], resource, codebase)
+                if new_resource:
+                    reference_licenses.extend(new_resource.licenses)
+                    reference_license_expressions.extend(new_resource.license_expressions)
+
+    licenses.extend(reference_licenses)
+    license_expressions.extend(reference_license_expressions)
+    codebase.save_resource(resource)
+    return resource
+
+
+def find_reference_licenses(referenced_filename, resource, codebase):
+    """
+    Returns the resource which references to referenced_filename if found
+    """
+    par = resource.parent(codebase)
+    
+    for child in par.children(codebase):
+        file_name = fileutils.file_name(child.path)
```

Comment:
we have a name attribute alright  in a Resource per https://github.com/nexB/commoncode/blob/b21f9f13b8d596be8246a48abf2c7f7d607cb5db/src/commoncode/resource.py#L966
But the name is not enough... `referenced_filename` can be a path too and you should just check that the path.endswith it

That said there are also many other things that would need to be handled (possibly in a second stage)... what if you have multiple `COPYING` files in a codebase and that `referenced_filename` is `"COPYING"`?
Which one would be selected in this case?
Which one is the correct one?

Reply:
Yes, we are only checking in the same directory as for now